### ISSUE TITLE
Cannot opt back into checked Sendable conformance

### ISSUE THREAD

**Issue by mattmassicotte** [apple/swift#74498]
### Description

I was surprised to see that I cannot opt back into a checked Sendable conformance to silence this warning.

### Reproduction

```swift
class A: @unchecked Sendable {
}

final class B: A, Sendable {
}
```

### Expected behavior

I would expect this code to be warning free.

### Environment

Apple Swift version 6.0-dev (LLVM 6e13b8be259f20f, Swift 372de4670b9cd55)
Target: arm64-apple-macosx14.0

**Comment by hborla** [comment]
Yeah, I don't know how to actually allow this, because the compiler cannot check anything that's inherited from the superclass, so you'd get a false sense of promise that your type really has a checked `Sendable` conformance.

**Comment by mattmassicotte** [comment]
My naive assumption was that the compiler could perform the same checking as it would if the superclass was checked-Sendable.
